AK/Assertions.h

```diff
@@ -20,3 +20,15 @@ static constexpr bool TODO = false;
 #define TODO_RISCV64() VERIFY(TODO) /* NOLINT(cert-dcl03-c,misc-static-assert) No, this can't be static_assert, it's a runtime check */
 #define TODO_PPC64() VERIFY(TODO)   /* NOLINT(cert-dcl03-c,misc-static-assert) No, this can't be static_assert, it's a runtime check */
 #define TODO_PPC() VERIFY(TODO)     /* NOLINT(cert-dcl03-c,misc-static-assert) No, this can't be static_assert, it's a runtime check */
+
+#ifdef NDEBUG
+extern "C" __attribute__((noreturn)) void ak_assertion_failed(char const*);
+#    define ASSERT(expr)                                                               \
+        (__builtin_expect(!(expr), 0)                                                  \
+                ? ak_assertion_failed(#expr " at " __FILE__ ":" __stringify(__LINE__)) \
+                : (void)0)
+#    define ASSERT_NOT_REACHED ASSERT(false) /* NOLINT(cert-dcl03-c,misc-static-assert) No, this can't be static_assert, it's a runtime check */
+#else
+#    define ASSERT(expr)
```

Comment:
The default build will no longer run these (at all), is anyone actually using the debug build? :thinking:

Reply:
I use it for debugging but not for performance work, don’t know about others.

Reply:
Then it seems kinda low-value, unless you're interested in this specific assertion, no one will notice it failing?
Maybe if we had CI running debug builds to ensure these assertions don't bitrot?

Reply:
I guess so.